fundamental-concepts/choices-role-pattern/2_api_to_choices.md

```diff
@@ -0,0 +1,110 @@
+First, some good news: The DAML ecosystem will generate the `createUser` endpoint automatically and
+there is nothing to do for us. In fact, it will provide a general `create` endpoint for all defined
+contract templates in your DAML model. For the provided HTTP JSON API this is the `v1/create`
+[endpoint](https://docs.daml.com/json-api/index.html#http-request). It will take the template name
+and its arguments as POST arguments to the call. In addition, there will also be an `archive`
+endpoint at `v1/archive` to archive an active contract and mark it inactive.
+
+Try to create `Alice`'s `User` contract by running a `create` request:
+
+```
+curl -X POST -H "Content-Type: application/json" -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwczovL2RhbWwuY29tL2xlZGdlci1hcGkiOnsibGVkZ2VySWQiOiJNeUxlZGdlciIsImFwcGxpY2F0aW9uSWQiOiJmb29iYXIiLCJhY3RBcyI6WyJBbGljZSJdfX0.VdDI96mw5hrfM5ZNxLyetSVwcD7XtLT4dIdHIOa9lcU' -d '{
+  "templateId": "User:User",
+  "payload": {
+    "username": "Alice",
+    "following": ["Bob"]
+  }}' localhost:7575/v1/create
+```{{execute T1}}
+
+1. The `Authorization` header authorizes the request with a dummy JWT token for `Alice`. You can
+   read more about authorization [here](https://docs.daml.com/json-api/index.html#with-authentication).
+1. The template identifier for the contract we want to create is specified in the `templateId`
+   field. In our case this is `User:User`.
+1. The arguments for the new contract are specified in the `payload`.
+1. We make `Alice` follow `Bob`, such that `Bob` can send her a message in the next command, by
+   adding him to the `following` list.
+
+The sandbox ledger returns the created contract ID in the response:
+
+```
+{"result":{"observers":["Bob"], "agreementText":"","contractId":"00dc73ac11bbe3181ffd74d1843a108ea2c625cdd5749624feeb2acbd670413fcf","key":"Alice","payload":{"username":"Alice","following":["Bob"]},"signatories":["Alice"],"templateId":"7ec600d1f061bd68a3aa3cf7b315d8d486cb3f6d05e8a341a2328039358888b8:User:User"},"status":200}
```

Comment:
I get one more field in the reply `exerciseResult`

{"result":{"exerciseResult":"0086e7201ee57141b9223215fefd772239ba0e53c542256ead6d05290dc519db86","events":[{"created":{"observers":[],"agreementText":"","payload":{"sender":"Bob","receiver":"Alice","content":"Hi Alice!"},"signatories":["Alice","Bob"],"contractId":"0086e7201ee57141b9223215fefd772239ba0e53c542256ead6d05290dc519db86","templateId":"7a08b0d6f54275b4d2be5275610ccdd24158a6705ff2be37900fe84c0abd3714:User:Message"}}]},"status":200}$

Reply:
Not quiet sure what you mean here. `exerciseResult` is always in the examples when you call `v1/exercise`